src/mmap.cc

```diff
@@ -85,9 +86,55 @@ NAN_METHOD(AlignedAlloc) {
 }
 
 
+NAN_METHOD(Sync) {
+  NanScope();
+  char* data = node::Buffer::Data(args[0]->ToObject());
+  size_t length =  node::Buffer::Length(args[0]->ToObject());
+
+  if (args.Length() > 1) {
+    // optional argument: offset
+    const size_t offset = args[1]->Uint32Value();
+    if (length <= offset) {
+      return NanThrowRangeError("Offset out of bounds");
+    }
+    else if (offset > 0 && (offset % getpagesize())) {
+      return NanThrowRangeError("Offset must be a multiple of page size");
+    }
+
+    data += offset;
+    length -= offset;
+
+  }
+
+  if (args.Length() > 2) {
+    // optional argument: length
+    const size_t range = args[2]->Uint32Value();
+    if (range < length) {
+      length = range;
+    }
+  }
+
+  int flags;
+  if (args.Length() > 3) {
+    // optional argument: flags
+    flags = args[3]->Uint32Value();
+  }
+  else {
+    flags = MS_SYNC;
+  }
+
+  if (msync(data, length, flags) == 0) {
```

Comment:
The `msync` man page says:

```
    The msync() system call writes modified whole pages back to the filesystem and updates the file modification time.  Only those pages containing addr and len-1 succeeding loca-
     tions will be examined.
```

If I get it right - to make this work properly you'll need to call `msync` on each page. You might want to use `getpagesize()` to figure out the size of the page, and do a loop, invoking `msync` on each iteration.